### PR TITLE
DEX-217 Add celery to houston for periodic and background tasks

### PR DESCRIPTION
**Pull request by karenc**
## Pull Request Overview

- Add celery to houston

  Run the celery worker by doing:
  
  ```
  celery -A app.extensions.celery.celery worker
  ```

- Fix link to "Project Structure" page in README

  It was pointing to a page that does not exist, change it to
  `docs/project_file_structure.md`.

- Add background and periodic tasks documentation

  Add example celery tasks to docs/background_tasks.md and link to it from
  the README

**Review Notes**

The first commit is what we want in order to use celery in houston.

But the other commits actually add example code to the application, I'm not sure if we want to merge this.  I just wanted to demonstrate how celery can be used.

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -136,9 +137,25 @@ def create_app(flask_config_name=None, config_override={}, testing=False, **kwar
     # Replace app.config (flask.Config) with our HoustonFlaskConfig version
     configure_using_houston_flask_config(app)
 
+    # Set up celery
+    redis_uri = f'redis://{app.config["REDIS_HOST"]}/0'
+    app.celery = Celery('houston', broker=redis_uri, backend=redis_uri)
```
Comment (hwindsor) [inline app/__init__.py:142]: Please help me understand this? Is this now listening on host/port of REDIS_HOST?
If so, how does this not clash with the tus code which is also using the same REDIS_HOST?

Comment (karenc) [reply app/__init__.py:142]: Yeah it is currently using the same redis database as tus.  We can change the `/0` to `/1` to use a different database in the same redis instance.

**Comment by karenc** [comment]
@colinwkingen Do we really need celery?  Or did we decide to just use cron jobs?

**Comment by colinwkingen** [comment]
@karenc I think we do want to have this for batching asset addition to gitlab. We will need a task queuing tool for image analysis jobs, which if my understanding of Celery's capabilities is correct this will be effective for also.

**Comment by karenc** [comment]
I think I'll move the example to the README

**Review by colinwkingen — APPROVED** [review]
Good to go.

**Pull request merged**